### PR TITLE
#21: Feature: Make main stat non-editable on the relic editing modal

### LINKED ISSUE #21: Feature: Make main stat non-editable on the relic editing modal
Currently both the enhance (e.g. +12) and main stat are editable

Goal: Level and grade should be enough to calculate the main stat so users shouldn't be able to edit main stats directly.

Other manual relic entry improvements are welcome too -> Auto setting main stat on Hands / Head piece for example

### PR DESCRIPTION
**Pull request by ronaldpereira**
- Makes the main stat non-editable on the relic editing modal. Did this by calculating it automatically with enhance level, grade and selected main stat.

image

image

image

image

- Makes the main stat dropdown to respect only the possible main stats for the selected relic part, including head to show only HP and hands only ATK.

image

image

image

image

image

image

image

Closes #21

### CODE REVIEW

**Comment by ronaldpereira** [comment]
The only thing that still bothers me is the fact that the only easy way I found to disable the edit on the main stat value was to use the disable, but this makes the font have a strange opacity. I accept suggestions on how to change this or if I should leave it as it is.

**Comment by ronaldpereira** [comment]
Now the only thing that bothers me is the main stat value style in the frontend.

With disabled prop it gives a better sense of "non-editable", but it also gives a strange opacity to it that makes it harder to read the value:

image

With readonly prop it doesn't differs from the other inputs, however it makes it more readable:

image

Let me know what you think

File: src/components/RelicModal.js [hunk 1]
```diff
@@ -249,7 +271,35 @@ export default function RelicModal(props) {
...
+    let grade = relicForm.getFieldValue('grade');
+
+    if (mainStatType != undefined && enhance != undefined && grade != undefined) {
+      const specialStats = [Constants.Stats.OHB, Constants.Stats.Physical_DMG, Constants.Stats.Physical_DMG, Constants.Stats.Fire_DMG, Constants.Stats.Ice_DMG, Constants.Stats.Lightning_DMG, Constants.Stats.Wind_DMG, Constants.Stats.Quantum_DMG, Constants.Stats.Imaginary_DMG];
+      const floorStats = [Constants.Stats.HP, Constants.Stats.ATK, Constants.Stats.SPD];
+
+      let mainStatValue = Constants.MainStatsValues[mainStatType][grade]['base'] + Constants.MainStatsValues[mainStatType][grade]['increment'] * enhance;
+
+      if (specialStats.includes(mainStatType)) { // Outgoing Healing Boost and elemental damage bonuses has a weird rounding with one decimal place
+        mainStatValue = Math.floor(mainStatValue * 10) / 10;
```
Comment (fribbels) [inline src/components/RelicModal.js:295]: @ronaldpereira Just for future reference I have a couple util functions that help for these cases

```
  precisionRound(number, precision = 8) {
    let factor = Math.pow(10, precision);
    return Math.round(number * factor) / factor;
  },
...
  truncate10ths: (x) => {
    return Math.floor(x * 10) / 10
  },
 ```

**Comment by fribbels** [comment]
Not blocking the PR but I'll do another pass to make the display values consistent. Ideally % stats always show 1 decimal 55.0% and flat stats are always rounded: 705 HP

File: src/components/RelicModal.js (hunk 1 above)
Comment (ronaldpereira) [reply src/components/RelicModal.js:295]: didn't see that when I was implementing it
we'll update this relic modal soon to implement more UI changes like we were discussing in discord, so I'll keep this in mind to use this functions, thanks!

**Pull request merged**